### ISSUE TITLE
Can I train with my own dataset?

### ISSUE THREAD

**Issue by Marchen0** [google-research/batch_rl#9]
Thanks for the great work first!
I have a bunch of data in **_(state, action, reward, next state)_** format. I try to understand how you guys parse the $store$_action_ckpt file in the code but I failed. It would be greatful if you could provide a way to train this model with my own dataset~

Thanks again

**Comment by agarwl** [comment]
Right, this is done via dopamine replay buffers which basically store the dataset in a specific format. You can try adding your data to a Dopamine buffer. A simpler way would be to create a tensorflow dataset out of your tuples or just sampling data directly:

```
minibatch ~ Sample(Dataset, batch_size)
model.train(minibatch)
```

If you have DQN implemented in your codebase, implementing methods from this repository  such as REM is a couple of more lines, so you can simply do that!

**Comment by agarwl** [comment]
@Marchen0 Is your issue resolved? If so, can you please close this issue? If not, can you describe what's the problem you are facing?

**Comment by fc1315** [comment]
@agarwl I have the same question regarding using my own offline data, not generated from Atari games. Looks like dopamine replay buffers have a very specific format and I need to change my data structure a lot. Could you give more details on how to implement 

> model.train(minibatch) 

directly? Thanks a lot!

**Comment by agarwl** [comment]
I think if you have data in the form of `(s, a, r, s')` tuples, you can create a tf/pytorch dataset (see [RL Unplugged](https://github.com/deepmind/deepmind-research/tree/master/rl_unplugged), or [D4RL](https://github.com/rail-berkeley/d4rl/blob/5120c12aec5089e2d57ffbc41420b111255b2735/d4rl/offline_env.py#L63)) containing these tuples. An even simpler (but less scalable) option is to read the dataset directly into memory using numpy arrays and samples minibatch tuples of `(s, a, r, s')` directly and pass it to the model for training.

That said, saving it to dopamine replay is also easy as you simply call the [`_store_transition`](https://github.com/google/dopamine/blob/master/dopamine/agents/dqn/dqn_agent.py#L370) function in a dopamine agent. So, if you have sequential data, you can simply read that data step by step and save it to disk using the [LoggedReplayBuffer](https://github.com/google-research/batch_rl/blob/f9b89f43add68b603ee562ebcf9dffcb6bf6f841/batch_rl/baselines/replay_memory/logged_replay_buffer.py#L36) using the  [`log_final_buffer`](https://github.com/google-research/batch_rl/blob/f9b89f43add68b603ee562ebcf9dffcb6bf6f841/batch_rl/baselines/replay_memory/logged_replay_buffer.py#L82) function.
